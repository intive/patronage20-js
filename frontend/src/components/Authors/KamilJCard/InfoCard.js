import React from 'react'

import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'

const useStyles = makeStyles({
  container: {
    textShadow: '-1px 0px 2px rgba(0,0,0,0.6)',
    color: 'white'
  },
  header: {
    padding: '1rem 0'
  }
})

export default function InfoCard () {
  const classes = useStyles()
  return (
    <Box className={classes.container}>
      <Typography className={classes.header} variant='h3'>
        <Box textAlign='center'>
          About me
        </Box>
      </Typography>
      <Typography className={classes.content} component='div'>
        <Box>
          Currently, I'm an aspiring full stack developer. I think of every challenge I approach as an opportunity to gain experience and to learn new things. Smart Home might be the biggest one I've ever encountered and so I am really eager to work on this project.
        </Box>
      </Typography>
    </Box>
  )
}
