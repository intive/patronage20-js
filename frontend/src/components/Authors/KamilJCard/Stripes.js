import React from 'react'
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles({
  stripe: {
    height: '100vh',
    borderRadius: 0
  }
})

export default function Stripes (props) {
  const classes = useStyles()

  return (
    <div>
      <Grid
        container
        direction='row'
        justify='flex-start'
        alignItems='center'
      >
        <Grid item xs={4}>
          <Paper elevation={0} style={{ backgroundColor: props.left }} className={classes.stripe} />
        </Grid>
        <Grid item xs={4}>
          <Paper elevation={0} style={{ backgroundColor: props.center }} className={classes.stripe} />
        </Grid>
        <Grid item xs={4}>
          <Paper elevation={0} style={{ backgroundColor: props.right }} className={classes.stripe} />
        </Grid>
      </Grid>
    </div>
  )
}
