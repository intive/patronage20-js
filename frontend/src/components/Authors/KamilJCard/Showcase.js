import React from 'react'

import InfoCard from './InfoCard'
import Links from './Links'

import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Avatar from '@material-ui/core/Avatar'

const useStyles = makeStyles({
  avatar: {
    boxShadow: '0px 0px 6px 2px rgba(0,0,0,0.75)',
    width: '180px',
    height: '180px'
  },
  container: {
    padding: 0,
    margin: 0,
    display: 'flex',
    alignItems: 'center',
    background: '#3badfc',
    height: '100vh'
  },
  center: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    color: 'white',
    textShadow: '-1px 0px 2px rgba(0,0,0,0.6)'
  }
})

export default function Showcase (props) {
  const classes = useStyles()
  return (
    <Box className={classes.container}>
      <Grid
        container
        spacing={0}
        direction='row'
        justify='center'
        alignItems='center'
      >
        <Grid item lg={6} xs={12}>
          <Typography className={classes.text} variant='h1'>
            <Box style={{ marginBottom: '0.5rem', fontSize: '3.3rem' }} textAlign='center'>Kamil Ja≈Çowiczor</Box>
          </Typography>
        </Grid>
        <Grid item lg={6} xs={12} className={classes.center}>
          <Avatar sizes='medium' className={classes.avatar} src={props.avatar} alt='avatar' />
        </Grid>
        <Grid item lg={6} md={10} xs={12}>
          <InfoCard style={{ height: '100%' }} />
        </Grid>
        <Grid item lg={6} md={10} xs={12}>
          <Links style={{ height: '100%' }} github={props.github} />
        </Grid>
      </Grid>
    </Box>
  )
}
