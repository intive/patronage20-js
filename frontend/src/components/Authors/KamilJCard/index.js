import KamilJCard from './KamilJCard.jsx'

export default KamilJCard
