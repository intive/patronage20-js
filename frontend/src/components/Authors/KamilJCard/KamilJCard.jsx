import React, { useEffect, useState } from 'react'
import axios from 'axios'

import Stripes from './Stripes'
import Showcase from './Showcase'

import { makeStyles, createMuiTheme, responsiveFontSizes, ThemeProvider } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import Grid from '@material-ui/core/Grid'

const useStyles = makeStyles({
  body: {
    margin: 0,
    padding: 0
  },
  root: {
    width: '100vw',
    height: '100vh',
    padding: 0,
    margin: 0
  },
  stripe: {
    height: '100vh',
    borderRadius: 0,
    background: '#9debff'
  }
})

export default function KamilJCard () {
  const classes = useStyles()
  document.body.className = classes.body
  const [dataState, setDataState] = useState([])

  useEffect(() => {
    axios.get('http://127.0.0.1:3000/api/v1/authors/3')
      .then(res => { setDataState(res.data) })
      .catch(err => { throw err })
  }, [])

  let theme = createMuiTheme()
  theme = responsiveFontSizes(theme)

  return (
    <Box className={classes.root}>
      <ThemeProvider theme={theme}>
        <Grid
          container
          spacing={0}
        >
          <Grid item xs={1}>
            <Stripes left='#0087e4' center='#0096fe' right='#3badfc' />
          </Grid>
          <Grid item xs={10}>
            <Showcase avatar={dataState.avatar} name={dataState.name} github={dataState.github} />
          </Grid>
          <Grid item xs={1}>
            <Stripes left='#3badfc' center='#59d4ff' right='#9debff' />
          </Grid>
        </Grid>
      </ThemeProvider>
    </Box>
  )
}
