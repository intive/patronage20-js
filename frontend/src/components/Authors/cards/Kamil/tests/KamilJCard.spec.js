/* globals describe, test, expect */

import React from 'react'
import TestRenderer from 'react-test-renderer'
import KamilJCard from '../KamilJCard'

describe('<KamilJCard />', () => {
  test('renders author KamilJCard component', () => {
    const root = TestRenderer.create(<KamilJCard />)
    expect(root.toJSON()).toMatchSnapshot()
  })
})
