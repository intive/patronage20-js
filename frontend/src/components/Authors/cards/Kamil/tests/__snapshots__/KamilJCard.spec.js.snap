// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KamilJCard /> renders author KamilJCard component 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-root-2"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#0087e4",
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#0096fe",
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#3badfc",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
    >
      <div
        className="MuiBox-root MuiBox-root-143 makeStyles-container-139"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
          >
            <h1
              className="MuiTypography-root makeStyles-text-141 MuiTypography-h1"
            >
              <div
                className="MuiBox-root MuiBox-root-174"
                style={
                  Object {
                    "fontSize": "3.3rem",
                    "marginBottom": "0.5rem",
                  }
                }
              >
                Kamil Ja≈Çowiczor
              </div>
            </h1>
          </div>
          <div
            className="MuiGrid-root makeStyles-center-140 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-138 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 MuiGrid-grid-lg-6"
          >
            <div
              className="MuiBox-root MuiBox-root-193 makeStyles-container-191"
            >
              <h3
                className="MuiTypography-root makeStyles-header-192 MuiTypography-h3"
              >
                <div
                  className="MuiBox-root MuiBox-root-194"
                >
                  About me
                </div>
              </h3>
              <div
                className="MuiTypography-root MuiTypography-body1"
              >
                <div
                  className="MuiBox-root MuiBox-root-195"
                >
                  Currently, I'm an aspiring full stack developer. I think of every challenge I approach as an opportunity to gain experience and to learn new things. Smart Home might be the biggest one I've ever encountered and so I am really eager to work on this project.
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 MuiGrid-grid-lg-6"
          >
            <div
              className="MuiBox-root MuiBox-root-199 makeStyles-container-196"
            >
              <div
                className="MuiBox-root MuiBox-root-200 makeStyles-content-197"
              >
                <a
                  className="makeStyles-link-198"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    style={
                      Object {
                        "fontSize": 55,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#3badfc",
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#59d4ff",
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              className="MuiPaper-root makeStyles-stripe-109 MuiPaper-elevation0 MuiPaper-rounded"
              style={
                Object {
                  "backgroundColor": "#9debff",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
